### ISSUE TITLE
check: does `torch.stack()` w/ a for loop detach gradients?

### ISSUE THREAD

**Issue by Jemoka** [alliegriffith/perez_reimplementation#13]
https://github.com/alliegriffith/perez_reimplementation/blob/e36436c3bc2e6664b79564a31dc5ecf2f55ae3c7/perez.py#L232-L238
